src/chart/util/setDefaults.js

```diff
@@ -0,0 +1,117 @@
+    var variables = ["id","major","minor","group","details"]
+    variables.forEach(function(varName){
+        chart.config.variables[varName] = chart.config.variables[varName] || defaultSettings.variables[varName]
+    })
+
+    //filters
+    chart.config.filters = chart.config.filters || defaultSettings.filters
+
+    //groups
+    chart.config.groups = chart.config.groups || defaultSettings.groups
+  
+    //defaults  
+    var defaults = ["maxPrevalence","totalCol","diffCol","prefTerms"]
+    chart.config.defaults = chart.config.defaults || {}
+    chart.config.defaults["maxPrevalence"] = chart.config.defaults["maxPrevalence"] || defaultSettings.defaults["maxPrevalence"]
+    chart.config.defaults["totalCol"] = chart.config.defaults["totalCol"]  != undefined ? 
+      chart.config.defaults["totalCol"]: 
+      defaultSettings.defaults["totalCol"]
+    chart.config.defaults["diffCol"] = chart.config.defaults["diffCol"]!= undefined?
+      chart.config.defaults["diffCol"]: 
+      defaultSettings.defaults["diffCol"]
+    chart.config.defaults["prefTerms"] = chart.config.defaults["prefTerms"]!=undefined?
+      chart.config.defaults["prefTerms"]:
+      defaultSettings.defaults["prefTerms"]
+
+
+    //plot settings
+    chart.config.plotSettings = chart.config.plotSettings || {}
+    var plotSettings = ["h","w","r", "margin","diffMargin"]
+    plotSettings.forEach(function(varName){
+        chart.config.plotSettings[varName] = chart.config.plotSettings[varName] || defaultSettings.plotSettings[varName]
+    })
+
+    ////////////////////////////////////////////////////////////
+    //Render single column if no group variable is specified. //
+    ////////////////////////////////////////////////////////////
+    if (!(chart.config.variables.group) || ['', ' '].indexOf(chart.config.variables.group) > -1) {
+        chart.config.variables.group = 'data_all';
+        chart.config.defaults.totalCol = false;
+        chart.config.groups = [{'key':'All'}];
+    }
+
+    ////////////////////////////////////////////////////
+    // Include all group levels if none are specified //
+    ////////////////////////////////////////////////////
+    var groups = d3.set(chart.raw_data.map(d => d[chart.config.variables.group])).values();
+    var groupsObject = groups.map(d => {return {'key': d}; });
+
+    if (!chart.config.groups || chart.config.groups.length === 0)
+        chart.config.groups = groupsObject
+            .sort((a,b) => a.key < b.key ? -1 : a.key > b.key ? 1 : 0);
+
+    //////////////////////////////////////////////////////////////
+    //Check that variables specified in settings exist in data. //
+    //////////////////////////////////////////////////////////////
+    for (var x in chart.config.variables) {
+        var varList = d3.keys(chart.raw_data[0]).concat('data_all');
+
+        if (varList.indexOf(chart.config.variables[x]) === -1) {
+            if (chart.config.variables[x] instanceof Array) {
```

Comment:
Add filters to chart.config.variables above, mapped to keep only the `value_col` property.